src/llmcompressor/transformers/finetune/data/flickr_30k.py

```diff
@@ -0,0 +1,62 @@
+from copy import deepcopy
+from typing import TYPE_CHECKING
+
+from llmcompressor.transformers.finetune.data import TextGenerationDataset
+from llmcompressor.typing import Processor
+
+if TYPE_CHECKING:
+    from llmcompressor.transformers import DataTrainingArguments as DataArgs
+
+
+@TextGenerationDataset.register(name="flickr", alias="flickr30k")
+class Flickr30K(TextGenerationDataset):
+    """
+    :param data_args: configuration settings for dataset loading
+    :param split: split from dataset to load, for instance `test` or `train[:5%]`
+    :param processor: processor or tokenizer to use on dataset
+    """
+
+    DEFAULT_CHAT_TEMPLATE = (
+        "{% for message in messages %}\n"
+        "{% if message['role'] == 'user' %}\n"
+        "{{ '<|user|>\n' + message['content'] + eos_token }}\n"
+        "{% elif message['role'] == 'system' %}\n"
+        "{{ '<|system|>\n' + message['content'] + eos_token }}\n"
+        "{% elif message['role'] == 'assistant' %}\n"
+        "{{ '<|assistant|>\n'  + message['content'] + eos_token }}\n"
+        "{% endif %}\n"
+        "{% if loop.last and add_generation_prompt %}\n"
+        "{{ '<|assistant|>' }}\n{% endif %}\n{% endfor %}"
+    )
+
+    def __init__(self, data_args: "DataArgs", split: str, processor: Processor):
+        data_args = deepcopy(data_args)
+        data_args.dataset = "lmms-lab/flickr30k"
+
+        super().__init__(data_args=data_args, split=split, processor=processor)
+
+        if (
+            self.tokenizer is not None
+            and getattr(self.tokenizer, "chat_template", None) is None
+        ):
+            # note that since tokenizer is a member of processor,
+            # this change affects processor.apply_chat_template
+            self.tokenizer.chat_template = self.DEFAULT_CHAT_TEMPLATE
```

Comment:
IMO we should try to remove any usage of default chat templates. We should require that the processor either has one registered, or that the user specifies one. Transformers has moved away from default chat templates and vLLM has as well due to bad behavior with models that don't recognize the special tokens.

Reply:
I'm not quite sure what that means in practice. For chat-based datasets, *some* function is required to convert the `messages` into `text`. Are you suggesting that said function avoid including special tokens?

Reply:
I'm saying we must either find a chat template in the tokenizer/processor or the user must provide one to override (we don't need to support this right now). If this isn't true, we should fail saying a chat template is required

This follows the behavior of transformers
```
>>> from transformers import AutoTokenizer
>>> tokenizer = AutoTokenizer.from_pretrained("facebook/opt-125m")
>>> input = tokenizer.apply_chat_template([{"role": "user", "content": "Hey, what's the temperature in Paris right now?"}])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mgoin/code/transformers/src/transformers/tokenization_utils_base.py", line 1616, in apply_chat_template
    chat_template = self.get_chat_template(chat_template, tools)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mgoin/code/transformers/src/transformers/tokenization_utils_base.py", line 1784, in get_chat_template
    raise ValueError(
ValueError: Cannot use chat template functions because tokenizer.chat_template is not set and no template argument was passed! For information about writing templates and setting the tokenizer.chat_template attribute, please see the documentation at https://huggingface.co/docs/transformers/main/en/chat_templating
```

Reply:
Cool, let's do that

Reply:
Addressed here: https://github.com/vllm-project/llm-compressor/pull/943/commits/e71f4e582cdbe81e3cc3f70847058a30fb1bbaf1

Reply:
Added back here: https://github.com/vllm-project/llm-compressor/pull/943/commits/bfefb58bca7313599b5171bf2abde706d702e80e

```
2024-12-23T17:50:11.581292-0500 | __init__ | WARNING - tokenizer.chat_template is not set, using default chat template for UltraChatDataset
```